code/hud/hudsquadmsg.cpp

```diff
@@ -1630,13 +1631,38 @@ void hud_squadmsg_type_select( )
 {
 	int k, i;
 
+	int num_order_types = NUM_COMM_ORDER_TYPES;
+
+	int lua_order_count = 0;
+
+	// Now get a list of all lua categories to add. Meow.
+	SCP_vector<SCP_string> lua_cat_list = ai_lua_get_general_order_categories();
+
+	// If we have too many categories then we have an oopsie!
+	if ((int)lua_cat_list.size() > (MAX_MENU_ITEMS - NUM_COMM_ORDER_TYPES)) {
+		Warning(LOCATION,
+			"Too many defined Lua General Order Categories! Any categories beyond %i will be ignored.",
+			MAX_MENU_ITEMS - NUM_COMM_ORDER_TYPES);
```

Comment:
Is it at all possible to validate this during parsing and not during display?

Reply:
Just have to move the relevant defines in to the header, I suppose.